konfera/models/schedule.py

```diff
@@ -23,3 +25,35 @@ class Schedule(KonferaModel):
 
     def __str__(self):
         return _('%(start)s (%(duration)s min)') % {'start': self.start, 'duration': self.duration}
+
+    def clean(self, *args, **kwargs):
+        # Only approved talk can be scheduled
+        if self.talk.status != 'approved':
+            raise ValidationError(_('You cannot schedule unapproved talks'))
+
+        # Event is related to location and location has room,
+        # make sure selected room in schedule belongs to Event's location
+        room = get_model('Room').objects.filter(location=event.location)
+        if location_room.first() != self.room:
+            raise ValidationError(_('The room that you have chosen is '
+                'not part of the scheduled room'))
```

Comment:
Add period to the end.

We have a 120 characters per line restriction, I think you can fit the whole sentence to one line. (but I didn't try it)